searchcore/src/tests/proton/reference/gid_to_lid_change_handler/gid_to_lid_change_handler_test.cpp

```diff
@@ -260,6 +260,38 @@ TEST_F("Test that pending removes are merged", Fixture)
     f.removeListeners("testdoc", {});
 }
 
+TEST_F("Test that out of order notifyRemoveDone is handled", Fixture)
+{
+    auto &stats = f.addStats();
+    auto listener = std::make_unique<MyListener>(stats, "test", "testdoc");
```

Comment:
Consider creating a fixture that has a single listener added and use this for the last 4 tests. This avoids the noise of dealing with the listener, which is not relevant for what is tested.